### PR TITLE
specs

### PR DESCRIPTION
**Pull request by dliangsta**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,6 @@
 ]
 
 EXTRAS_REQUIRE = {
-    "union": ["typeguard"],
```
Comment (dliangsta) [inline setup.py:16]: basically making union a requirement, not an extra

File: marshmallow_dataclass/__init__.py [hunk 2]
```diff
@@ -367,7 +358,9 @@ def _field_by_type(
 ) -> Optional[Type[marshmallow.fields.Field]]:
     return (
         base_schema and base_schema.TYPE_MAPPING.get(typ)
-    ) or marshmallow.Schema.TYPE_MAPPING.get(typ)
+    ) or marshmallow.Schema.TYPE_MAPPING.get(typ) or (
+        any(kw in str(typ) for kw in ["Serializable", "Mapping", "Dict"]) and dict
+    )
```
Comment (dliangsta) [inline marshmallow_dataclass/__init__.py]: A hack that translates SerializableAttributes, Mappings, and Dicts into dicts. pretty nifty

File: marshmallow_dataclass/__init__.py [hunk 3]
```diff
@@ -74,14 +72,6 @@ class User:
...
 
 
-class BaseSchema(marshmallow.Schema):
-    TYPE_MAPPING = {
-        SerializableAttribute: marshmallow.fields.Mapping,
-        SerializableAttributeTag: marshmallow.fields.Mapping,
-        TypedDict: marshmallow.fields.Mapping,
-    }
-
-
```
Comment (dliangsta) [inline marshmallow_dataclass/__init__.py:84]: there would be a circular dependency if blib/specs needed marshmallow_dataclass, but marshmalow_dataclass imported these types

File: marshmallow_dataclass/__init__.py [hunk 4]
```diff
@@ -60,8 +60,6 @@ class User:
 
 from typing import TypedDict
 
-from blib.attributes import SerializableAttribute, SerializableAttributeTag
```
Comment (dliangsta) [inline marshmallow_dataclass/__init__.py:63]: there would be a circular dependency if blib/specs needed marshmallow_dataclass, but marshmalow_dataclass imported these types

File: marshmallow_dataclass/__init__.py [hunk 5]
```diff
@@ -367,6 +358,8 @@ def _field_by_type(
 ) -> Optional[Type[marshmallow.fields.Field]]:
     return (
         base_schema and base_schema.TYPE_MAPPING.get(typ)
+    ) or (
+        any(kw in str(typ) for kw in ["Serializable", "Mapping", "Dict"]) and dict
```
Comment (dliangsta) [inline marshmallow_dataclass/__init__.py:362]: kind of a hack but works pretty well. the issue was basically that we had a lot of types inheriting from TypedDict that marshmallow_dataclass didn't recognize as dicts

File: marshmallow_dataclass/__init__.py [hunk 6]
```diff
@@ -475,13 +468,13 @@ def field_for_schema(
...
             return list_type(child_type, **metadata)
-        if origin in (tuple, Tuple):
+        elif origin in (tuple, Tuple):
             children = tuple(
                 field_for_schema(arg, base_schema=base_schema) for arg in arguments
             )
             tuple_type = type_mapping.get(Tuple, marshmallow.fields.Tuple)
             return tuple_type(children, **metadata)
-        elif origin in (dict, Dict):
+        elif origin in (dict, Dict):# or any(kw in str(type) for kw in ["Serializable", "Mapping", "Dict"]):
```
Comment (gnedivad) [inline marshmallow_dataclass/__init__.py]: Is this necessary?

**Review by gnedivad — APPROVED** [review]

File: marshmallow_dataclass/__init__.py (hunk 6 above)
Comment (dliangsta) [reply marshmallow_dataclass/__init__.py]: nope

**Pull request merged**